config/initializers/default_host.rb

```diff
@@ -16,4 +16,8 @@
   Rails.application.routes.default_url_options.merge!( server_options )
   ActionMailer::Base.default_url_options.merge!( server_options )
   ApplicationController.default_url_options = server_options
+
+  # Required for rails 6+
+  # See https://blog.saeloun.com/2019/10/31/rails-6-adds-guard-against-dns-rebinding-attacks.html
+  Rails.application.config.hosts << server_options[:host]
```

Comment:
This is the substantive change.  The rest of the PR is fixing the tests to ensure we're using the host configured here.